packages/jupyter-ai/src/hooks/use-copy.ts

```diff
@@ -38,15 +47,19 @@ export type UseCopyReturn = {
 const DEFAULT_LABELS_BY_COPY_STATUS: Record<CopyStatus, string> = {
   [CopyStatus.None]: 'Copy to clipboard',
   [CopyStatus.Copying]: 'Copying…',
-  [CopyStatus.Copied]: 'Copied!'
+  [CopyStatus.Copied]: 'Copied!',
+  [CopyStatus.Disabled]: 'Copy to clipboard disabled in insecure context'
 };
 
 /**
  * Hook that provides a function to copy a string to a clipboard and manages
  * related UI state. Should be used by any button that intends to copy text.
  */
 export function useCopy(props?: UseCopyProps): UseCopyReturn {
-  const [copyStatus, setCopyStatus] = useState<CopyStatus>(CopyStatus.None);
+  const isCopyDisabled = navigator.clipboard === undefined;
+  const [copyStatus, setCopyStatus] = useState<CopyStatus>(
+    isCopyDisabled ? CopyStatus.Disabled : CopyStatus.None
+  );
   const timeoutId = useRef<number | null>(null);
 
   const copy = useCallback(
```

Comment:
The `copy()` function also should be modified to do nothing when `copyStatus === CopyStatus.Disabled`. This keeps `copyStatus` set to `CopyStatus.Disabled` even if `copy()` is called somehow.